libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -567,17 +590,11 @@ constructor(
     }
 
     private fun reroute() {
-        ifNonNull(tripSession.getEnhancedLocation()) { location ->
-            val optionsRebuilt = AdjustedRouteOptionsProvider.getRouteOptions(
-                directionsSession,
-                tripSession,
-                location
-            ) ?: return
-            directionsSession.requestRoutes(
-                optionsRebuilt,
-                null
-            )
-        }
+        rerouteController.reroute(object : RerouteController.RoutesCallback{
```

Comment:
reroute is only called once per reroute, so the retry logic inside RerouteController won't be hit multiple times right?

Reply:
yeah, that's right! I'm thinking one more point: 
- user offroute 
- reroute controller start fetching new route 
- user returns to route 
- [reroute controller have to interrupt route fetched]

Last point is not implemented, but I think should be 🤔